server/src/public/routes/platform/customer/GetDbDumpHandlerWithFilter.ts

```diff
@@ -0,0 +1,584 @@
+import type { Request, Response } from 'express';
+import Pg from 'pg';
+import { to as copyTo } from 'pg-copy-streams';
+import isUUID from 'validator/lib/isUUID.js';
+import {
+  forwardHandlerExceptionsToNext,
+  ApiCallerError,
+} from 'server/src/public/routes/platform/util.ts';
+import { anonymousLogger } from 'server/src/logging/Logger.ts';
+import type { DatabaseConfig } from 'server/src/util/readReplicaDatabase.ts';
+import env from 'server/src/config/Env.ts';
+import { getReadReplicaDbConfigFromEnv } from 'server/src/util/readReplicaDatabase.ts';
+
+async function GetDbDumpHandlerWithFilter(req: Request, res: Response) {
+  const customerID = req.customerID;
+  if (!customerID) {
+    throw new ApiCallerError('invalid_access_token');
+  }
+
+  res.statusCode = 200;
+  res.setHeader('Content-type', 'text/plain');
+
+  const appIDToIgnore = req.body.appID;
+  if (!!appIDToIgnore && !isUUID.default(appIDToIgnore)) {
+    throw new ApiCallerError('invalid_request');
+  }
```

Comment:
Added this bit. Then passed appIDToIgnore through the various levels until we get to the raw SQL query -- at that point, added a "AND appID != appIDToIgnore"